src/predictor/gpu_predictor.cu

```diff
@@ -500,25 +463,27 @@ class GPUPredictor : public xgboost::Predictor {
                  const std::vector<std::shared_ptr<DMatrix>>& cache) override {
     Predictor::Configure(cfg, cache);
 
-    GPUSet devices = GPUSet::All(learner_param_->gpu_id, learner_param_->n_gpus);
-    ConfigureShards(devices);
+    int device = learner_param_->gpu_id;
+    if (device >= 0) {
+      ConfigureShards(device);
+    }
   }
 
  private:
   /*! \brief Re configure shards when GPUSet is changed. */
-  void ConfigureShards(GPUSet devices) {
-    if (devices_ == devices) return;
+  void ConfigureShards(int device) {
+    if (device_ == device) return;
 
-    devices_ = devices;
+    device_ = device;
     shards_.clear();
-    shards_.resize(devices_.Size());
+    shards_.resize(1);
     dh::ExecuteIndexShards(&shards_, [=](size_t i, DeviceShard& shard){
-        shard.Init(devices_.DeviceId(i));
+        shard.Init(device_);
       });
   }
 
   std::vector<DeviceShard> shards_;
```

Comment:
similar to earlier and applies to all: this need not be an array anymore

Reply:
Done.